src/sentry/static/sentry/app/views/organizationDiscover/conditions/condition.jsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from 'grid-emotion';
+
+import SelectControl from 'app/components/forms/selectControl';
+
+import {getInternal, getExternal, isValidCondition} from './utils';
+import {CONDITION_OPERATORS} from '../data';
+
+export default class Condition extends React.Component {
+  static propTypes = {
+    value: PropTypes.array,
+    onChange: PropTypes.func,
+    columns: PropTypes.array,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedColumn: null,
+      selectedOperator: null,
+    };
+  }
+
+  focus() {
+    this.select.focus();
+  }
+
+  handleChange = option => {
+    const external = getExternal(option.value, this.props.columns);
+
+    if (isValidCondition(external, this.props.columns)) {
+      this.props.onChange(external);
+      return;
+    }
+
+    if (new Set(this.props.columns.map(({name}) => name)).has(external[0])) {
+      this.setState({selectedColumn: external[0]}, this.focus);
+    }
+
+    if (new Set(CONDITION_OPERATORS).has(external[1])) {
```

Comment:
why not just `Array.includes`?

Reply:
force of habit, and using sets elsewhere here